### PR TITLE
46684033 pull request COMP3710

### PR DESCRIPTION
**Pull request by NickWWWW1**
Hi,

This pull request contains my attempt on Task 7, which is to create a classifier based on the Siamese network [10] to classify either Alzheimer’s disease. 

The best result I got was 80.5% on the test set.

This pull request contains the following:

1. modules.py: The source code for Siamese neural network+ classifier model architecture

2. dataset.py: The source code for data preprocessing, and creating data loaders

3. train.py: The source code for training both Siamese neural network and the respective classifier

4. predict.py: The source code for evaluating the performance of the classifier

5. README.md: Project Documentation

6. Images_for_README: a folder container for all images for README.md 

Thanks,

Man Lik Nicholas Wong

46684033

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Very good
- Clear flow of information
- Algortihm and training details explained well
- Relevant plots for loss, accuracy and 2D manifolds provided
- Presents all results for different model combinations and settings

Commit messages: Good, detailed

Code: 
- good design
- minimal comments

Functionality/Performance: 
- siamese network seems to be doing a good job of separating the two classes according to the manifold figures
- classifier achieves >80% accuracy for test set

General comments:
- algorithm solves the problem apporpriately with expected performance

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
